scalar.go

```diff
@@ -31,21 +32,48 @@ type Identifier struct {
 }
 
 type IdentifierInput struct {
-	Id    *ID     `graphql:"id" json:"id,omitempty" yaml:"id,omitempty"`
-	Alias *string `graphql:"alias" json:"alias,omitempty" yaml:"alias,omitempty"`
+	id    *ID     `json:"id,omitempty" yaml:"id,omitempty"`
+	alias *string `json:"alias,omitempty" yaml:"alias,omitempty"`
+}
+
+func (i IdentifierInput) MarshalJSON() ([]byte, error) {
+	// this case can only occur by using an EmptyIdentifier()
+	if i.id == nil && i.alias == nil {
+		return []byte("null"), nil
+	}
+	return json.Marshal(struct {
+		Id    *ID     `json:"id,omitempty"`
+		Alias *string `json:"alias,omitempty"`
+	}{
+		Id:    i.id,
+		Alias: i.alias,
+	})
+}
+
+func (i *IdentifierInput) Id() *ID {
+	return i.id
+}
+
+func (i *IdentifierInput) Alias() *string {
+	return i.alias
 }
 
 func NewIdentifier(value string) *IdentifierInput {
 	if IsID(value) {
 		return &IdentifierInput{
-			Id: NewID(value),
+			id: NewID(value),
 		}
 	}
 	return &IdentifierInput{
-		Alias: NewString(value),
+		alias: NewString(value),
 	}
 }
 
+func EmptyIdentifier() *IdentifierInput {
```

Comment:
This is a necessary and special identifier that can be used to unset a field.

For example, you can pass `EmptyIdentifier()` in a `ServiceUpdateInput` to unset the `Owner` of a Service. 

Or, you can just not set a value for it and it the Service's owner will be untouched because of `omitempty`.